test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -80,19 +81,24 @@ private void assertCommandBehavior(String inputCommand,
         //Execute the command
         CommandResult r = logic.execute(inputCommand);
 
+        //Parse the String input command to a Command object
+        Command currentCommand = new Parser().parseCommand(inputCommand);
+
         //Confirm the result contains the right data
         assertEquals(expectedMessage, r.feedbackToUser);
         assertEquals(r.getRelevantPersons().isPresent(), isRelevantPersonsExpected);
-        if(isRelevantPersonsExpected){
+        if (isRelevantPersonsExpected) {
             assertEquals(lastShownList, r.getRelevantPersons().get());
         }
 
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
-    }
 
+        if (currentCommand.isMutating()) {
```

Comment:
This change doesn’t help much because it does not confirm that the saving happens _only_ for mutating commands. Tests will pass even if you don’t have this change. To be complete, you should have an `else` branch that confirms the file was not updated for non-mutating commands (e.g. by checking the last updated time of the file).